src/replication.c

```diff
@@ -580,6 +604,52 @@ handle_read_resp(spec_t *spec, replica_t *replica, zvol_io_hdr_t *io_rsp, void *
 	return 0;
 }
 
+#define handle_consistency_met() {\
+	rcommq_ptr->status = 1; \
+	MTX_LOCK(&spec->rcommonq_mtx); \
+	TAILQ_REMOVE(&spec->rcommon_waitq, rcommq_ptr, wait_cmd_next); \
+	if (rcommq_ptr->acks_recvd != rcommq_ptr->copies_sent) { \
+		rcommq_ptr->state = CMD_ENQUEUED_TO_PENDINGQ; \
+		TAILQ_INSERT_TAIL(&spec->rcommon_pendingq, rcommq_ptr, pending_cmd_next); \
+	} else { \
+		for (i=1; i < rcommq_ptr->iovcnt + 1; i++) { \
+			xfree(rcommq_ptr->iov[i].iov_base); \
+		} \
+		rcommq_ptr->state = CMD_EXECUTION_COMPLETED; \
+	} \
+	MTX_UNLOCK(&spec->rcommonq_mtx); \
+	signal_luworker(); \
+}
+
+#define handle_all_resp_recvd() { \
+	MTX_LOCK(&spec->rcommonq_mtx); \
+	TAILQ_REMOVE(&spec->rcommon_pendingq, rcommq_ptr, pending_cmd_next); \
```

Comment:
Here, we come if acks_recv < CF or acks_recv > CF. For second case, this code looks fine. There may be case where all resp recvd, but, consistency factor is not reached. So, we need to check the state of rcommand_t to remove from pendingq/waitq. we need to wakeup luworker in case of first one.